web/src/message_view.js

```diff
@@ -442,7 +442,12 @@ export function show(raw_terms, opts) {
                     );
 
                     if (adjusted_terms === null) {
-                        blueslip.error("adjusted_terms impossibly null");
+                        // If the stream was deleted, there will be no change in
+                        // terms. We cannot narrow to the message in this case.
+                        // User likely starred the message and the message's
+                        // stream got deleted. When user clicks on the message
+                        // we reach here. Returning will keep user narrowed to
+                        // the starred messages which what we want here.
                         return;
```

Comment:
Is the only scenario that we'll get here that the channel was deleted? How do we know?

Reply:
I guess other case would be user doesn't have permission to access stream. Which doesn't make any difference here?